src/nautilus-server/src/apps/seal-example/README.md

```diff
@@ -179,39 +200,64 @@ Encoded seal responses:
 `-t`: Threshold used for encryption. 
 `-n`: The network of the key servers you are using.
 
-### Step 4: Complete Secret Loading
+### Step 4: Complete Key Load
+
+This step is done in the host that the enclave runs in, that can communicate to the enclave via 3001. If it returns OK, the enclave decrypts and caches the Seal keys in memory. Replace `<ENCODED_SEAL_RESPONSES>` with the output from Step 3.
 
-This step is done in the host that the enclave runs in, that can communicate to the enclave via 3001. The server decrypts the first secret and initializes it as the `SEAL_API_KEY`. It also returns the rest as dummy secrets as an example, remove as needed. 
 
 ```bash
-curl -X POST http://localhost:3001/seal/complete_parameter_load \
+curl -X POST http://localhost:3001/admin/complete_seal_key_load \
   -H "Content-Type: application/json" \
   -d '{
-    "encrypted_objects": "<ENCRYPTED_OBJECT>",
     "seal_responses": "<ENCODED_SEAL_RESPONSES>"
   }'
 
-{"dummy_secrets":[[1,1]]}
+# Expected response:
+{"status":"OK"}
 ```
 
-In this call, the enclave uses its ephemeral secret key to decrypt key shares and performs threshold decryption to recover the secrets. Then the enclave finishes the bootstrap phase by storing the decrypted secret `SEAL_API_KEY` in memory.
+### Step 5: Provision Weather API Key
+
+This step is done in the host that the enclave runs in, that can communicate to the enclave via port 3001. Replace `<ENCRYPTED_OBJECT>` with the output from Step 1.
+
+In this call, the enclave uses the cached keys from Step 4 to decrypt the encrypted weather API key. This endpoint is application specific and replace or add more if needed. Repeat step 1 to encrypt other data to the enclave object ID and provision them to the enclave with an endpoint. 
 
-### Step 5: Use the Service
+```bash
+curl -X POST http://localhost:3001/admin/provision_weather_api_key \
+  -H "Content-Type: application/json" \
+  -d '{
+    "encrypted_object": "<ENCRYPTED_OBJECT>"
+  }'
+
+# Expected response:
+{"status":"OK"}
+```
+
+### Step 6: Use the Service
 
 Now the enclave server is fully functional to process data. 
 
 ```bash
 curl -H 'Content-Type: application/json' -d '{"payload": { "location": "San Francisco"}}' -X POST http://<PUBLIC_IP>:3000/process_data
 
+# Example response: 
 {"response":{"intent":0,"timestamp_ms":1755805500000,"data":{"location":"San Francisco","temperature":18}},"signature":"4587c11eafe8e78c766c745c9f89b3bb7fd1a914d6381921e8d7d9822ddc9556966932df1c037e23bedc21f369f6edc66c1b8af019778eb6b1ec1ee7f324e801"}
 ```
 
 ## Handle Multiple Secrets
 
-In step 1, pass in a list of secrets and a list of IDs to get an encoded list of encrypted objects. At step 2, pass in all IDs from step 1 to `ids` in an array so the returned `FetchKeyRequest` is constructed over all IDs. Step 3 and step 4 are unchanged. 
+One can encrypt many secrets to the enclave ID. Repeat step 1 with any data, using the same package ID and the same enclave object ID. 
```

Comment:
```suggestion
Since Seal uses public key encryption, one can encrypt many secrets to the enclave ID. Repeat step 1 with any data, using the same package ID and the same enclave object ID. 
```